### DISCUSSION TITLE
How to call apis in jars in extension

### DISCUSSION THREAD

**Discussion by travis1111** [microsoft/vscode-discussions#1088] [Extension Development QnA] [answered]
Hi, I am wondering what is the best way to call methods in jar files in vscode extension?  is there any vscode api for that? or use child process in nodejs?

Thanks

**Comment by abrilurena** [comment, marked as answer]
Hi, @travis1111! I hope you're doing great, I found this, please let me know if it works for you: When working with Java JAR files in a Visual Studio Code (VSCode) extension, you have a few options. Let’s explore them:

1. Using the Java Dependency Viewer Extension:

- First, ensure you have the Java Dependency Viewer extension installed in VSCode.
- Open your Java project folder (not just a single file) in VSCode.
- Inside the left panel, you’ll find “Java Dependencies.” Click the “+” symbol to add custom JAR files.
- This extension will manage the classpath and allow you to use classes from the JAR files directly in your code.

2. Manually Adding JARs to the Classpath:

- If you’re not using Maven or Gradle for dependency management, you can manually add JAR files to the classpath.
- Create a folder named “lib” inside your project (if not already present).
- Place your JAR files inside this “lib” folder.
- Open the .vscode/settings.json file (create one if it doesn’t exist) and add the JAR files to the java.project.referencedLibraries setting.

3. Using Maven or Gradle:

- If you’re using Maven or Gradle, they handle dependency management automatically.
- Add your JAR dependencies to the respective pom.xml (Maven) or build.gradle (Gradle) files.
- Build your project, and the dependencies will be resolved.

Remember that VSCode is primarily a text editor, not a full-fledged IDE. However, with the right extensions and configuration, you can effectively work with Java projects in VSCode.
